modules/nanopolish/1.0/nanopolish.smk

```diff
@@ -0,0 +1,266 @@
+        installed = str(rules._install_nanopolish.output.complete)
+    params:
+         nano = CFG["inputs"]["nanopolish_exec"] + "/nanopolish"
+    threads:
+        CFG["threads"]["nanopolish"] 
+    log:
+        stderr = CFG["logs"]["meth_calls"] + "{seq_type}--{genome_build}/{sample_id}/{chrom}.meth_calls.stderr.log"     
+    resources: 
+        mem_mb = CFG["mem_mb"]["meth_calls"]          
+    output:
+        calls = CFG["dirs"]["meth_calls"] + "{seq_type}--{genome_build}/{sample_id}/chromosomes/{chrom}.calls.tsv.gz"    
+
+    shell:
+        "{params.nano} call-methylation -t {threads} -r {input.fastq} -b {input.bam} -g {input.fasta} -q cpg -w {wildcards.chrom} | gzip > {output.calls} "
+
+
+rule _nanopolish_meth_freq:
+        input: 
+            calls = str(rules._nanopolish_meth_calls.output.calls)
+        output: 
+            freq = CFG["dirs"]["meth_freq"] + "{seq_type}--{genome_build}/{sample_id}/chromosomes/{chrom}.frequency.tsv.gz"
+        params:
+            nano = CFG["inputs"]["nanopolish_exec"] + "/scripts/calculate_methylation_frequency.py" 
+        log:
+            stderr = CFG["logs"]["meth_freq"] + "{seq_type}--{genome_build}/{sample_id}/{chrom}.meth_freq.stderr.log"
+        resources: 
+            mem_mb = CFG["mem_mb"]["meth_freq"] 
+        shell:
+            "{params.nano} -s {input.calls} | gzip > {output.freq}"
+
+
+def _nanopolish_get_chr_meth_calls(wildcards):
+    CFG = config["lcr-modules"]["nanopolish"]
+    chrs = checkpoints._promethion_input_chrs.get(**wildcards).output.chrs
+    with open(chrs) as file:
+        chrs = file.read().rstrip("\n").split("\n")
+    calls = expand(
+        CFG["dirs"]["meth_calls"] + "{{seq_type}}--{{genome_build}}/{{sample_id}}/chromosomes/{chrom}.calls.tsv.gz",
+        chrom = chrs
+    )
+    return(calls)
+
+
+def _nanopolish_get_chr_meth_freq(wildcards):
+    CFG = config["lcr-modules"]["nanopolish"]
+    chrs = checkpoints._promethion_input_chrs.get(**wildcards).output.chrs
+    with open(chrs) as file:
+        chrs = file.read().rstrip("\n").split("\n")
+    freqs = expand(
+        CFG["dirs"]["meth_freq"] + "{{seq_type}}--{{genome_build}}/{{sample_id}}/chromosomes/{chrom}.frequency.tsv.gz",
+        chrom = chrs
+    )
+    return(freqs)
+
+
+rule _merge_nanopolish_calls:
+    input:
+        calls = _nanopolish_get_chr_meth_calls
+    output:
+        calls = CFG["dirs"]["merged_nanopolish_calls"] + "{seq_type}--{genome_build}/{sample_id}.calls.tsv.gz"
+    resources: 
+        mem_mb = CFG["mem_mb"]["merged_nanopolish_calls"] 
+    log:
+        stderr = CFG["logs"]["merged_nanopolish_calls"] + "{seq_type}--{genome_build}/{sample_id}/merged_nanopolish_calls.stderr.log"               
+    shell:
+        op.as_one_line("""
+        zcat {input.calls} | sort -V -k1,1 -k2,2 |
+        cat <(grep "chromosome" | head -n1) <(grep -v "chromosome") | 
+        gzip >  {output.calls}
```

Comment:
Can you add a redirect to write the stderr of this command to your log file?

Reply:
This applies to all shell bash commands

Reply:
Redirected stderr to log file and removed logs from merge rules since there wouldn't be any stderr generated from bash cat commands